qiskit/synthesis/discrete_basis/solovay_kitaev.py

```diff
@@ -0,0 +1,257 @@
+        return v_n1.dot(w_n1).dot(v_n1.adjoint()).dot(w_n1.adjoint()).dot(u_n1)
+
+    def find_basic_approximation(self, sequence: GateSequence) -> Gate:
+        """Finds gate in ``self._basic_approximations`` that best represents ``sequence``.
+
+        Args:
+            sequence: The gate to find the approximation to.
+
+        Returns:
+            Gate in basic approximations that is closest to ``sequence``.
+        """
+        # TODO explore using a k-d tree here
+
+        def key(x):
+            return np.linalg.norm(np.subtract(x.product, sequence.product))
+
+        best = min(self.basic_approximations, key=key)
+        return best
+
+
+class SolovayKitaev(TransformationPass):
+    r"""Approximately decompose 1q gates to a discrete basis using the Solovay-Kitaev algorithm.
+
+    See :mod:`qiskit.transpiler.synthesis.solovay_kitaev` for more information.
+
+    """
+
+    def __init__(
+        self,
+        recursion_degree: int = 3,
+        basic_approximations: str | dict[str, np.ndarray] | None = None,
+    ) -> None:
+        """
+        Args:
+            recursion_degree: The recursion depth for the Solovay-Kitaev algorithm.
+                A larger recursion depth increases the accuracy and length of the
+                decomposition.
+            basic_approximations: The basic approximations for the finding the best discrete
+                decomposition at the root of the recursion. If a string, it specifies the ``.npy``
+                file to load the approximations from. If a dictionary, it contains
+                ``{label: SO(3)-matrix}`` pairs. If None, a default based on the H, T and Tdg gates
+                up to combinations of depth 10 is generated.
+        """
+        super().__init__()
+        self.recursion_degree = recursion_degree
+        self._sk = SolovayKitaevDecomposition(basic_approximations)
+
+    def run(self, dag: DAGCircuit) -> DAGCircuit:
+        """Run the ``SolovayKitaev`` pass on `dag`.
+
+        Args:
+            dag: The input dag.
+
+        Returns:
+            Output dag with 1q gates synthesized in the discrete target basis.
+
+        Raises:
+            TranspilerError: if a gates does not have to_matrix
+        """
+        for node in dag.op_nodes():
+            if not node.op.num_qubits == 1:
+                continue  # ignore all non-single qubit gates
+
+            if not hasattr(node.op, "to_matrix"):
+                raise TranspilerError(
+                    "SolovayKitaev does not support gate without "
+                    f"to_matrix method: {node.op.name}"
+                )
+
+            matrix = node.op.to_matrix()
+
+            # call solovay kitaev
+            approximation = self._sk.run(matrix, self.recursion_degree)
+
+            # convert to a dag and replace the gate by the approximation
+            substitute = circuit_to_dag(approximation)
+            dag.substitute_node_with_dag(node, substitute)
+
+        return dag
```

Comment:
I think this can be deleted because there is a duplicate in `qiskit/transpiler/passes/synthesis`?

Reply:
whoops my bad, I forgot to delete it after moving it! Thanks for catching that!